### ISSUE TITLE
Fan speed above 100% G14 2020

### ISSUE THREAD

**Issue by marcelomijas** [seerge/g-helper#36]
Hi! There is a problem when I apply custom fan curves in GHelper:
- When CPU fan profile is set between 60-70%, GHelper reports that the fan is at 97% (which do not correspond, but I am aware of the difficulty of reading fan speeds...)
- **When GPU fan profile is set above 70%, GHelper reports that the fan is at 112%** (that should be impossible).

Generally, the fan speed reported is always ~30% above the fans profile specification at the current temp.

![imagen](https://user-images.githubusercontent.com/62946623/222222386-d4eae9a5-2f2c-4d25-a13e-02b6fdec733a.png)

Information: GHelper v0.10, laptop model G14 2020 (Ryzen 7 4800HS and 2060 Max-Q), Windows 10 Home 22H2.

I tried the new Power Limits (PPT) feature, but it does not limit the power of my CPU (or system).

Also, GPU Modes Eco and Standard does not seem to have a difference in the GPU and iGPU usage when switching.

Thanks for the amazing software! I ditched Armoury Crate days ago and I am not coming back!

**Comment by MrEleventy** [comment]
> Also, GPU Modes Eco and Standard does not seem to have a difference in the GPU and iGPU usage when switching.

According to this [reddit post](https://www.reddit.com/r/ZephyrusG14/comments/11bkr8l/light_tool_to_control_your_laptop_without_bloat/ja12f01/), the 2020 model doesn't support GPU mode switching. It requires bios support for it to work.

**Comment by seerge** [comment]
You list multiple issues under one :)

1. dGPU switch is not supported by 2020 hardware, so has no effect. Best thing you can do - is just adjust windows power plan setting to enforce integrated graphics on battery (and would be still have "recommendation" character for apps)

2. A way to set PPTs is something i have discovered by myself for 2022 model, it's probably also not supported (in the same way) on older models. You can use ryzenadj or similar tool instead tho

3. Asus bios has quite strange way of interpreting fan settings (also if you set them from Armory!). If you set fan speed to something lower than 30% (like 10), but non 0 -  it will still turn them on, and run at least on 2000rpm .

4.  Fan % is currently shown directly from what sensor in bios reports, and is relative to something like 6000rpm max. Probably on 2020 model either fan can spin faster (so ratio is > 1) or sensor reading is weird :D I will look into that and see if there is any smarter way to report ... may be simply report RPM instead of %

**Comment by seerge** [comment]
I have added display of RPM next to % in new build 
https://github.com/seerge/g-helper/releases/tag/v0.12

Also fan editor now shows RPM instead of %, and is "calibrated" to real fan speeds :) (calibrated base on my own 2022 model, your mileage may vary)

**Comment by marcelomijas** [comment]
Sorry for the multple issues listing under one.

1-4. Now I understand. Thank you!

I am trying v0.12 and I am happy for the change, in this case I prefer RPM instead of % (fans above 100% still occurs on GPU, but it's fine with me). 

Also, I wonder if you know the software LibreHardwareMonitor (Open Source), it might help you to get stats about wattage, temps and etc. I am complementing it with GHelper, it uses very few resources.
https://github.com/LibreHardwareMonitor/LibreHardwareMonitor

Again, thanks!

**Comment by seerge** [comment]
Yes, I was actually using LibreHardwareMonitor in earlier version of GHelper, but then I have found a way to get CPU temp / battery readings in an easier way and stopped.

Main reason is that LibreHardwareMonitor for majority of readings requires admin permissions, and currently app is running admin free (which is a big advantage for average user imho). 

I'm also not a big fan of turning GHelper into a hardcore monitoring tool, as there are great tools that do it already :)

**Comment by seerge** [comment]
Can you try CPU PPTs in https://github.com/seerge/g-helper/releases/tag/v0.13 ?

**Comment by marcelomijas** [comment]
Hi, tried v0.13. CPU PPTs still does not work and GHelper version is still saying 0.12.0 (I checked that I downloaded and exectuted v0.13). Also, this is the first version that Windows Defender marks as a threat: Trojan:Script/Wacatac.H!ml
